Models/ResourceChangesModel.h

```diff
@@ -0,0 +1,32 @@
+#ifndef RESOURCECHANGESMODEL_H
+#define RESOURCECHANGESMODEL_H
+
+#include "ResourceModelMap.h"
+
+#include <QObject>
+#include <QAbstractListModel>
+#include <QMap>
+
+enum class ResChange : int {
+  Added,
+  Modified,
+  Renamed,
+  Removed,
+  Reverted
+};
+
+class ResourceChangesModel : public QAbstractListModel
```

Comment:
Is it a common practice to create separate models to use only as an index over some display model? My understanding is that we generally don't create a model for something we don't intend to display in the UI. And you're not planning to display all resources as a list, right? If you're just adding this model class as something that listens for resource changes and updates disk, I don't think that's right. I think it can just be a normal manager class. @RobertBColton please confirm or deny.

Reply:
this is displayed tho